source/grid/grid_in.cc

```diff
@@ -2811,6 +2815,420 @@ GridIn<dim, spacedim>::read_assimp(const std::string &filename,
 #endif
 }
 
+namespace
+{
+  /**
+   * Find the reference cell type from the number of vertices.
+   */
+  template <int dim>
```

Comment:
Is it important that it is templated?

Reply:
No

Reply:
Actually, now that I look again, it does need to be templated since a quadrilateral and a tetrahedron have the same number of vertices.

Reply:
I know that number of vertices is not unique to the reference-cell type, but the combination of number of vertices and dimension. But this does not mean that the function has to be templated; you can pass in dim as a normal parameter.

I have a similar function here:

https://github.com/dealii/dealii/blob/cd6047cf8a54efac208a38d4d45fa9d407ca876d/include/deal.II/grid/connectivity.h#L1577-L1597

Here I created a 2D array of reference-cell types. Unfortunately, the formatting is horrible.

Reply:
That could work.

Reply:
I refactored both places into a common function - see 90df7baffa4a9c080c281e5736afa36658b5c107.